### PR TITLE
passage: use getopt from nix on darwin

### PR DESCRIPTION
**Pull request by phaer**
This patches a hack from upstream which hard-codes a bunch of paths to find gnu-getopt (as opposed to darwins getopt in /usr/bin/getopt).

This lets nix-built passage fail on systems which don't have one of those already installed (i.e. from homebrew).

As upstream seemingly doesn't provide a way to override this easily yet, we do it ourselves.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/passage/default.nix [hunk 1]
```diff
@@ -22,15 +23,21 @@ stdenv.mkDerivation {
     sha256 = "1val8wl9kzlxj4i1rrh2iiyf97w9akffvr0idvbkdb09hfzz4lz8";
   };
 
+  patches = lib.optional stdenv.isDarwin [ ./darwin-getopt-path.patch ];
```
Comment (wegank) [inline pkgs/tools/security/passage/default.nix:26]: Try doing something like

```suggestion
  patches = [
    (substituteAll {
      src = ./darwin-getopt-path.patch;
      GETOPT = "${getopt}/bin/getopt";
    })
  ];
```

Comment (phaer) [reply pkgs/tools/security/passage/default.nix:26]: Ah right, we can actually patch this on all platforms if we want as it's just unused on others.

File: pkgs/tools/security/passage/default.nix [hunk 2]
```diff
@@ -22,6 +24,13 @@ stdenv.mkDerivation {
     sha256 = "1val8wl9kzlxj4i1rrh2iiyf97w9akffvr0idvbkdb09hfzz4lz8";
   };
 
+  patches = lib.optional stdenv.isDarwin [
```
Comment (wegank) [inline pkgs/tools/security/passage/default.nix:27]:
```suggestion
  patches = [
```

Or, in the worst case, `lib.optionals stdenv.isDarwin [`.

**Comment by phaer** [comment]
@wegank Oh no, the last suggested change broke it, substitution does not work like this.

**Pull request merged**